### DISCUSSION TITLE
Installing the plugin Metafields through Cloudron , it doesnt want to install

### DISCUSSION THREAD

**Discussion by timraub** [kimai/Paid-plugins#130] [Questions & Answers]
I'm tried to add the plugin and follow the directions on their site and when i go to use the command 

bin/console kimai:bundle:metafields:install

It still tells me ...

There are no commands defined in the "kimai:bundle:XYZ" namespace.

Anyway you can help me? I paid for the plugin.. I just wanna get it going please...

**Comment by kevinpapst** [comment]
> Anyway you can help me? I paid for the plugin.. I just wanna get it going please...

Understand.
BUT: The plugin works fine if you install it the proposed way. 
I cannot support every possible environment. 
I don't know Cloudron and how it works. 

But I know there is documentation on how to install plugins:
https://docs.cloudron.io/apps/kimai/

If the documentation doesn't help, post in their specific Kimai forum at:
https://forum.cloudron.io/category/45/kimai

**Reply by timraub** [reply]
Its ok , im trying again without it

**Reply by kevinpapst** [reply]
Reading their docs it seems to be almost like the "pure" installation. 
Did it not work?

**Reply by timraub** [reply]
I just did a fresh ubuntu install .. no cloudron 

I got this error for the metafields plugin on clearing the cache before i did the install command 

```
PHP Fatal error:  Uncaught TypeError: array_key_exists(): Argument #2 ($array) must be of type array, null given in /var/www/kimai/src/Plugin/PluginMetadata.php:40
Stack trace:
#0 /var/www/kimai/src/Kernel.php(112): App\Plugin\PluginMetadata->__construct()
#1 /var/www/kimai/src/Kernel.php(77): App\Kernel->getBundleClasses()
#2 /var/www/kimai/vendor/symfony/http-kernel/Kernel.php(346): App\Kernel->registerBundles()
#3 /var/www/kimai/vendor/symfony/http-kernel/Kernel.php(762): Symfony\Component\HttpKernel\Kernel->initializeBundles()
#4 /var/www/kimai/vendor/symfony/http-kernel/Kernel.php(126): Symfony\Component\HttpKernel\Kernel->preBoot()
#5 /var/www/kimai/vendor/symfony/framework-bundle/Console/Application.php(190): Symfony\Component\HttpKernel\Kernel->boot()
#6 /var/www/kimai/vendor/symfony/framework-bundle/Console/Application.php(72): Symfony\Bundle\FrameworkBundle\Console\Application->registerCommands()
#7 /var/www/kimai/vendor/symfony/console/Application.php(175): Symfony\Bundle\FrameworkBundle\Console\Application->doRun()
#8 /var/www/kimai/vendor/symfony/runtime/Runner/Symfony/ConsoleApplicationRunner.php(49): Symfony\Component\Console\Application->run()
#9 /var/www/kimai/vendor/autoload_runtime.php(29): Symfony\Component\Runtime\Runner\Symfony\ConsoleApplicationRunner->run()
#10 /var/www/kimai/bin/console(13): require_once('...')
#11 {main}
  thrown in /var/www/kimai/src/Plugin/PluginMetadata.php on line 40
```

I installed the plugins to 
/var/www/kimai/var/plugins/

then i did the clear cache command and got the error above

**Reply by timraub** [reply]
Im using 22.04 ubuntu server , php 8.1 , and latest version of kimai 2.11.0

**Comment by pbksol** [comment]
Kimai2 Version 2 doesn’t support PHP 7.1. 

See requirements at: https://www.kimai.org/documentation/installation.html

**Reply by timraub** [reply]
IM on 8.1 it seems

PHP 8.1.2-1ubuntu2.14 (cli) (built: Aug 18 2023 11:41:11) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.2, Copyright (c) Zend Technologies
    with Zend OPcache v8.1.2-1ubuntu2.14, Copyright (c), by Zend Technologies
radmin@kimai2:/var/www/kimai$

**Reply by timraub** [reply]
Am i supposed to copy one plugin at a time or can i copy all the plugins and then run the cache and then run updates for each one? I bought 4 of your plugins

**Reply by timraub** [reply]
Holy hell im dumb, its one at a time...and now its working great

**Reply by kevinpapst** [reply]
No ,that doesn't matter, you can install as many plugins at once as you want.
I guess you forgot to change the directory name from one of the plugins?
Anyway, great to read that it works now 👍

**Reply by timraub** [reply]
Yes, thank you ! Thank you very much and i hope you have a great day
